chia/wallet/wallet_node.py

```diff
@@ -643,10 +638,23 @@ async def receive_and_validate(inner_state: CoinState, inner_idx: int):
                 return False
             while num_concurrent_tasks >= target_concurrent_tasks:
                 await asyncio.sleep(1)
+                if self._shut_down:
+                    break
             all_tasks.append(asyncio.create_task(receive_and_validate(potential_state, idx)))
             num_concurrent_tasks += 1
 
-        await asyncio.gather(*all_tasks)
+        if not self._shut_down:
+            await asyncio.gather(*all_tasks)
+        else:
+            for task in all_tasks:
+                task.cancel()
```

Comment:
isn't this kind of dangerous? Can we wait for the tasks to end, since most of them will be quick anyways?  I don't want half the DB operations to finish for example

Reply:
If there is such a hazard then our db locking/transactions are wrong I would think?  For example, I'm doing this because of a pile of errors that lead me to look at this.  https://github.com/Chia-Network/chia-blockchain/runs/5269700934?check_suite_focus=true

No, my changes so far do not fix the flaky test failure, I don't think, but it seemed a useful step to quiet stuff down a bit.